image_match/elasticsearch_driver.py

```diff
@@ -70,27 +71,26 @@ def search_single_record(self, rec, pre_filter=None):
                               size=self.size,
                               timeout=self.timeout)['hits']['hits']
 
-        sigs = np.array([x['_source']['signature'] for x in res])
-
-        if sigs.size == 0:
+        if len(res) == 0:
             return []
 
-        dists = normalized_distance(sigs, np.array(signature))
-
         formatted_res = [{'id': x['_id'],
                           'score': x['_score'],
                           'metadata': x['_source'].get('metadata'),
                           'path': x['_source'].get('url', x['_source'].get('path'))}
                          for x in res]
 
-        for i, row in enumerate(formatted_res):
-            row['dist'] = dists[i]
-        formatted_res = filter(lambda y: y['dist'] < self.distance_cutoff, formatted_res)
+        formatted_res = filter(lambda y: y['score'] > self.score_cutoff, formatted_res)
 
         return formatted_res
 
     def insert_single_record(self, rec, refresh_after=False):
         rec['timestamp'] = datetime.now()
+
+        # Don't store signature in index
+        if 'signature' in rec:
```

Comment:
We should include a setting in the class constructor whether we want to store the signature or not.  The default should be yes, to keep it backwards compatible.  I can work on this.

Reply:
I can look at it sometime in the near future